test_pytest_rerunfailures.py

```diff
@@ -306,3 +312,65 @@ def test_pass(self, setup_fixture, param):
                 assert param""")
     result = testdir.runpytest('--reruns', '1')
     assert_outcomes(result, passed=3, rerun=1)
+
+
+def test_rerun_on_module_fixture_with_reruns(testdir):
+    """
+    Case: Module scope fixture is not re-executed when class scope fixture throwing error on the first execution
+    for parametrized test
+    """
+    testdir.makepyfile("""
+        import pytest
+        
+        class Execution:
+            count = 0
```

Comment:
I like those two tests and I see that you have this pattern of execution number, but I seems more pythonic to me to use a global boolean in this case instead of a singleton class to store it. But that is on you.

Reply:
Thank you for the suggestion. Will redo this.